core/src/test/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionsServiceSpec.scala

```diff
@@ -0,0 +1,471 @@
+import org.broadinstitute.dsde.rawls.metrics.RawlsStatsDTestUtils
+import org.broadinstitute.dsde.rawls.mock._
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.rawls.openam.MockUserInfoDirectivesWithUser
+import org.broadinstitute.dsde.rawls.resourcebuffer.ResourceBufferServiceImpl
+import org.broadinstitute.dsde.rawls.serviceperimeter.ServicePerimeterServiceImpl
+import org.broadinstitute.dsde.rawls.user.UserService
+import org.broadinstitute.dsde.rawls.util.MockitoTestUtils
+import org.broadinstitute.dsde.rawls.webservice._
+import org.broadinstitute.dsde.rawls.workspace.{MultiCloudWorkspaceAclManager, MultiCloudWorkspaceService, RawlsWorkspaceAclManager, WorkspaceRepository, WorkspaceService}
+import org.broadinstitute.dsde.rawls.{ RawlsExceptionWithErrorReport, RawlsTestUtils, TestExecutionContext}
+import org.broadinstitute.dsde.workbench.dataaccess.{NotificationDAO, PubSubNotificationDAO}
+import org.broadinstitute.dsde.workbench.google.mock.{MockGoogleBigQueryDAO, MockGoogleIamDAO, MockGoogleStorageDAO}
+import org.broadinstitute.dsde.workbench.model.google.{BigQueryDatasetName, BigQueryTableName, GoogleProject}
+import org.mockito.ArgumentMatchers._
+import org.mockito.Mockito._
+import org.mockito.{Mockito}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.{BeforeAndAfterAll, OptionValues}
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.jdk.DurationConverters.JavaDurationOps
+import scala.language.postfixOps
+
+
+class SubmissionsServiceSpec extends AnyFlatSpec
+  with ScalatestRouteTest
+  with Matchers
+  with TestDriverComponent
+  with RawlsTestUtils
+  with Eventually
+  with MockitoTestUtils
+  with RawlsStatsDTestUtils
+  with BeforeAndAfterAll
+  with TableDrivenPropertyChecks
+  with OptionValues {
+
+  import driver.api._
+
+  val workspace = Workspace(
+    testData.wsName.namespace,
+    testData.wsName.name,
+    "aWorkspaceId",
+    "aBucket",
+    Some("workflow-collection"),
+    currentTime(),
+    currentTime(),
+    "test",
+    Map.empty
+  )
+
+  val mockServer = RemoteServicesMockServer()
+
+  val leonardoDAO: MockLeonardoDAO = new MockLeonardoDAO()
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    mockServer.startServer()
+  }
+
+  override def afterAll(): Unit = {
+    mockServer.stopServer
+    super.afterAll()
+  }
+
+  // noinspection TypeAnnotation,NameBooleanParameters,ConvertibleToMethodValue,UnitMethodIsParameterless
+  class TestApiService(dataSource: SlickDataSource, val user: RawlsUser)(implicit
+                                                                        override val executionContext: ExecutionContext
+  ) extends WorkspaceApiService
+    with MethodConfigApiService
```

Comment:
This file is a lot bigger than what was removed. Did you add more test coverage, or is it all setup code (that can't be removed from the original test)?

Reply:
It's setup code.
Right now, there's some places where some of the state setup for individual tests is using the real code paths in other services, so it wasn't really safe to just go ripping things out (yet).